src/odig.ml

```diff
@@ -0,0 +1,62 @@
+open Import
+
+type t = Opam.t * Opam.Switch.t
+
+type of_opam_error = Odig_not_installed
+
+let of_opam (opam, switch) =
+  let open Promise.Syntax in
+  let+ is_installed = Opam.has_package opam switch "odig" in
+  if is_installed then
+    Ok (opam, switch)
+  else
+    Error Odig_not_installed
+
+let cmd_ouput (opam, switch) args =
+  let cmd = Opam.exec opam switch ~args in
+  Cmd.output cmd
+
+let conf t key =
+  let open Promise.Syntax in
+  let+ odig_conf_result = cmd_ouput t [ "odig"; "conf" ] in
+  match odig_conf_result with
+  | Error _ -> None
+  | Ok conf ->
+    let lines = String.split_lines conf in
+    let is_whitespace_or_single_quote c =
+      Char.equal c '\'' || Char.is_whitespace c
+    in
+    List.map lines ~f:(String.rsplit2 ~on:':')
+    |> List.filter_opt
+    |> List.find_map ~f:(fun (k, v) ->
+           if String.equal k key then
+             Some
+               (String.strip v ~drop:is_whitespace_or_single_quote
+               |> Path.of_string)
+           else
+             None)
```

Comment:
We may not need `conf` fn at all since we have `odig cache path`, but if we do:
we don't need to rsplit2 each line to find a particular key-value pair, I'd split on-demand and minimize unnecessary computations. Or if we want to query `odig conf` several times, I'd create a map from k-v pairs and return it

Reply:
I made the changes to use `odig cache path` (see above) so that code doesn't exist anymore